### PR TITLE
OBG 904: Expose Datasafe encryption settings in configuration file

### PR DESCRIPTION
**Pull request by ivaniuminov**
https://github.com/adorsys/open-banking-gateway/issues/904

### CODE REVIEW

File: opba-banking-protocol-facade/src/main/java/de/adorsys/opba/protocol/facade/config/encryption/DatasafeConfig.java [hunk 1]
```diff
@@ -102,6 +110,24 @@ public FintechConsentSpecSecureStorage fintechUserDatasafeServices(
...
+    private EncryptionConfig buildEncryptionConfig() {
+        return datasafeConfigRepository.findAll().stream()
+                       .findFirst()
+                       .map(e -> mapToEncryptionConfig(e.getConfig()))
+                       .orElseGet(this::encryptionConfig);
+    }
+
+    @SneakyThrows
+    private EncryptionConfig mapToEncryptionConfig(byte[] data) {
+        return mapper.readValue(new String(data), EncryptionConfig.class);
```
Comment (valb3r) [inline opba-banking-protocol-facade/src/main/java/de/adorsys/opba/protocol/facade/config/encryption/DatasafeConfig.java:128]: ObjectMapper can read directly from bytes, no need to do `new String(data)`

File: opba-embedded-starter/src/main/resources/application.yml [hunk 2]
```diff
@@ -144,6 +144,38 @@ facade:
         cipher-algo: AES256_CBC
         key-algo: RSA
         len: 2048
+  # Datasafe security configuration
+  datasafe:
+    # Which keystore to use and how to encrypt keys in it
+    key-store:
+      type: BCFKS
+      encryption-algo: AES256_KWP
```
Comment (valb3r) [inline opba-embedded-starter/src/main/resources/application.yml:152]: Use AES-128 as default everywhere

File: opba-banking-protocol-facade/src/main/java/de/adorsys/opba/protocol/facade/config/encryption/DatasafeConfig.java [hunk 3]
```diff
@@ -102,6 +110,24 @@ public FintechConsentSpecSecureStorage fintechUserDatasafeServices(
         );
     }
 
+    @Bean
+    @ConfigurationProperties(prefix = "facade.datasafe")
+    public EncryptionConfig encryptionConfig() {
+        return EncryptionConfig.builder().build();
+    }
+
+    private EncryptionConfig buildEncryptionConfig() {
```
Comment (valb3r) [inline opba-banking-protocol-facade/src/main/java/de/adorsys/opba/protocol/facade/config/encryption/DatasafeConfig.java:119]: Add log statement at INFO level to indicate from where (DB or YML) encryption configuration came

File: opba-embedded-starter/src/main/resources/application.yml (hunk 2 above)
Comment (ivaniuminov) [reply opba-embedded-starter/src/main/resources/application.yml]: AES-128 was used for  "encryption-algo" property in all 3 property-files

File: opba-banking-protocol-facade/src/main/java/de/adorsys/opba/protocol/facade/config/encryption/DatasafeConfig.java (hunk 1 above)
Comment (ivaniuminov) [reply opba-banking-protocol-facade/src/main/java/de/adorsys/opba/protocol/facade/config/encryption/DatasafeConfig.java]: Done, thanks

File: opba-banking-protocol-facade/src/main/java/de/adorsys/opba/protocol/facade/config/encryption/DatasafeConfig.java (hunk 3 above)
Comment (ivaniuminov) [reply opba-banking-protocol-facade/src/main/java/de/adorsys/opba/protocol/facade/config/encryption/DatasafeConfig.java:121]: Logging was added

File: opba-banking-protocol-facade/src/main/java/de/adorsys/opba/protocol/facade/config/encryption/DatasafeConfiger.java [hunk 4]
```diff
@@ -30,16 +35,18 @@
 
 import static de.adorsys.opba.protocol.facade.config.ConfigConst.FACADE_CONFIG_PREFIX;
 
+@Slf4j
 @Configuration
 @RequiredArgsConstructor
-public class DatasafeConfig {
+public class DatasafeConfiger {
```
Comment (valb3r) [inline opba-banking-protocol-facade/src/main/java/de/adorsys/opba/protocol/facade/config/encryption/DatasafeConfiger.java:41]: Typo `DatasafeConfiger` -> `DatasafeConfigurer`

Comment (ivaniuminov) [reply opba-banking-protocol-facade/src/main/java/de/adorsys/opba/protocol/facade/config/encryption/DatasafeConfiger.java]: Done, thanks

File: opba-protocols/xs2a-protocol/src/main/resources/application.yml [hunk 5]
```diff
@@ -27,6 +27,40 @@ spring:
...
+        mac-algo: HmacSHA3_512
+        password-keys-algo: PBEWithHmacSHA256AndAES_128
+      # Which keys to create in keystore
+      keys:
+        enc-key-number: 1
+        sign-key-number: 1
+        # Secret key creation
+        secret:
+          algo: AES
+          size: 128
```
Comment (valb3r) [inline opba-protocols/xs2a-protocol/src/main/resources/application.yml:50]: Replace AES to 128 in other places:
AES256_CBC

Comment (ivaniuminov) [reply opba-protocols/xs2a-protocol/src/main/resources/application.yml:50]: All 3 property files changed to 
psu:
      key-pair:
        cipher-algo: AES128_CBC

**Review by valb3r — APPROVED** [review]

**Pull request merged**